pkg/controller/controller_instance_test.go

```diff
@@ -215,7 +215,63 @@ func TestReconcileInstanceNonExistentServicePlan(t *testing.T) {
 	}
 }
 
-func TestReconcileInstanceWithParameters(t *testing.T) {
+func TestReconcileInstanceWithInvalidParameters(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, fakeBrokerClient, testController, sharedInformers := newTestController(t, noFakeActions())
+
+	sharedInformers.Brokers().Informer().GetStore().Add(getTestBroker())
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(getTestServiceClass())
+
+	instance := getTestInstance()
+
+	// Secret with invalid JSON
+	invalidSecret := &v1.Secret{
+		Data: map[string][]byte{
+			testSecretKey: []byte("invalid json"),
```

Comment:
> +1 to map<string,string> with an escape hatch of a json blob for the other/"as is" case.

Works for me.

I was arguing against duck typing and for JSON encoding in case I wasn't clear. But I'm also OK treating values always as strings for multi-key secrets.